pkg/manifests/manifests.go

```diff
@@ -767,6 +768,19 @@ func (f *Factory) KubeStateMetricsDeployment(enableCRSMetrics bool) (*appsv1.Dep
 			if enableCRSMetrics {
 				d.Spec.Template.Spec.Containers[i].Args = append(container.Args, flagCRSConfigFile)
 			}
+			additionalAllowList := f.config.ClusterMonitoringConfiguration.KubeStateMetricsConfig.AdditionalLabelsAllowList
+			if additionalAllowList != nil && *additionalAllowList != "" {
+				allowListValidator := ksmpkg.LabelsAllowList{}
+				err = allowListValidator.Set(*additionalAllowList)
+				if err != nil {
+					return nil, fmt.Errorf("error parsing allowlist: %v", err)
+				}
```

Comment:
I see, the bump happens for this PR specifically, and not the `main` since, we pull KSM in which specifies a higher version. This is blocked until we move to `1.24.x`.

```
[cluster-monitoring-operator] GOTOOLCHAIN=go1.23.0 go mod tidy                                                                       2e7fc5398  ✭
go: k8s.io/kube-state-metrics/v2@v2.16.0 requires go >= 1.24.0 (running go 1.23.0; GOTOOLCHAIN=go1.23.0)
```

Reply:
Post go1.24 update, bumping dependencies reveals a missing `k8s.io/api/resource/v1beta2` package for the commit pinned in the `replace` directive. Upon updating it, it seems to clash with certain vendored dependencies, at which point, I believe it's simpler to copy the validation function from KSM here.